File: 1331. Rank Transform of an Array/solution.java
class Solution {
    public int[] arrayRankTransform(int[] arr) {

        int n = arr.length;
        HashMap<Integer,Integer>hm = new HashMap<>();
        int nums[] = new int[n];

        for(int i=0;i<n;i++)
            nums[i] = arr[i];

        Arrays.sort(nums);
        int rank = 1;

        for(int i=0;i<n;i++){
            if(!hm.containsKey(nums[i])){
                hm.put(nums[i],rank);
                rank++;
            }
        }

        for(int i=0;i<n;i++)
            nums[i] = hm.get(arr[i]);

        return nums;
        
    }
}

File: 1331. Rank Transform of an Array/solution.js
/**
 * @param {number[]} arr
 * @return {number[]}
 */
var arrayRankTransform = function(arr) {

    const n = arr.length;
    const mp = new Map();
    const nums = [...arr];

    nums.sort((a, b) => a - b); 
    let rank = 1;

    for (let num of nums) {
        if (!mp.has(num)) {
            mp.set(num, rank);
            rank++;
        }
    }

    return arr.map(num => mp.get(num));
};